master/internal/resourcemanagers/config.go

```diff
@@ -0,0 +1,64 @@
+package resourcemanagers
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/union"
+)
+
+// Config hosts configuration fields of the scheduler.
+type Config struct {
+	Type             string                  `json:"type"`
+	Fit              string                  `json:"fit"`
+	ResourceProvider *ResourceProviderConfig `json:"resource_provider"`
+}
+
+// ResourceProviderConfig hosts configuration fields for the resource provider.
+type ResourceProviderConfig struct {
+	DefaultRPConfig    *DefaultResourceProviderConfig   `union:"type,default" json:"-"`
+	KubernetesRPConfig *KubernetesResourceManagerConfig `union:"type,kubernetes" json:"-"`
```

Comment:
why is one of thse of RP and one of these and RM?

Reply:
I reused the struct of `KubernetesResourceProviderConfig` to be `KubernetesResourceManagerConfig` since they have the same fields. Both `DefaultResourceProviderConfig` and `ResourceProviderConfig` should be removed when we don't maintain `scheduler` field in the future.